### PR TITLE
non-core subproject status changes

### PR DESCRIPTION
**Pull request by mikebrow**
Address #36

Merge Approval Requires 2/3 of the current committers (8 votes)
   - [x] @AkihiroSuda
   - [ ] @crosbymichael
   - [x] @dmcgowan
   - [x] @estesp
   - [ ] @stevvooe
   - [x] @mikebrow
   - [x] @fuweid
   - [x] @mxpv
   - [x] @dims
   - [ ] @kevpar
   - [x] @kzys
   - [x] @samuelkarp

Signed-off-by: Mike Brown <brownwm@us.ibm.com>

### CODE REVIEW

File: GOVERNANCE.md [hunk 1]
```diff
@@ -168,6 +168,28 @@ maintainers—the core maintainers of containerd will not be appended to each su
...
+
+The proposal should include some general information on support, releases,
+stability, downstream impacts, steps necessary to effect the change, and any
+additional detail useful for the containerd maintainers to understand the scope and
+nature of the project status change. GitHub issue is the easiest way to provide
+this proposal.
+
+The existing maintainers are given fourteen days to discuss the project change,
+raise objections and cast their vote. Project changes of this type must be approved
+by 2/3 of the current core committers.
```
[the remark this answers is not in the document]
Comment (mikebrow) [reply GOVERNANCE.md:187]: > * Do existing non-core maintainers get a binding vote?
> * For non-core maintainers who are not core committers, can they continue to help maintain the project after it becomes a core project?

For promoting and merging WDYT about - "The proposal should include consideration for any changes needed/desired to the current core committers and reviewers. For example, provide link(s) to vote(s) or a proposed set of changes to occur in concert." 

Removing/archiving indicate some change of direction or reduction in interest to maintaining the project. I think we would do the right thing here and have discussions with the non-core maintainers first and those discussions should/will sway the vote. WDYT about - "The removing/archiving proposal should include open discussion and consideration for the viewpoints of the current non-core committers, reviewers, and downstream users."


Such considerations, are why I proposed to extend the typical length of time for the vote.

[the remark this answers is not in the document]
Comment (samuelkarp) [reply GOVERNANCE.md:187]: > Do existing non-core maintainers get a binding vote?

I think based on your reply and re-reading this again, the answer is no?

> For non-core maintainers who are not core committers, can they continue to help maintain the project after it becomes a core project?

This also sounds like no, unless there are other (potentially coordinated) changes under the existing governance rules around adding maintainers.

----

Thinking about whether non-core committers should get a binding vote: we could require 2/3 of both the changed non-core project committers _and_ 2/3 of core committers to approve, but that would mean project status changes have a higher barrier than new projects to be added.  On the other hand, non-core maintainers should have a voice here.  What about wording like this?

```suggestion
The existing maintainers are given fourteen days to discuss the project change,
raise objections and cast their vote. Project changes of this type must be approved
by 2/3 of the current core committers and core committers are encouraged to consider the opinions and objections of the existing non-core project maintainers.

Project status changes do not automatically change the status of any non-core project maintainers.  Project status change proposals may be accompanied by additional proposals to change individual maintainer status, subject to the existing rules set forth in this document.
```

**Review by mxpv — APPROVED** [review]

**Review by dims — APPROVED** [review]
LGTM

**Review by dmcgowan — APPROVED** [review]

File: GOVERNANCE.md [hunk 2]
```diff
@@ -168,6 +168,30 @@ maintainers—the core maintainers of containerd will not be appended to each su
...
+stability, downstream impacts, steps necessary to effect the change, and any
+additional detail useful for the containerd maintainers to understand the scope and
+nature of the project status change. GitHub issue is the easiest way to provide
+this proposal.
+
+The existing maintainers are given fourteen days to discuss the project change,
+raise objections and cast their vote. Project changes of this type must be approved
+by 2/3 of the current core committers and core committers are encouraged to consider the opinions and objections of the existing non-core project maintainers.
+
+Project status changes do not automatically change the status of any non-core project maintainers.  Project status change proposals may be accompanied by additional proposals to change individual maintainer status, subject to the existing rules set forth in this document.
```
Comment (samuelkarp) [inline GOVERNANCE.md:189]: nit: wrapping is not right here

**Review by samuelkarp — APPROVED** [review]

File: GOVERNANCE.md (hunk 2 above)
Comment (mikebrow) [reply GOVERNANCE.md:189]: yeah md doesn't care, the view gets updated when it is loaded.. more a convention to insert extra line breaks or not

**Review by kzys — APPROVED** [review]

**Review by fuweid — APPROVED** [review]
LGTM

**Review by AkihiroSuda — APPROVED** [review]

**Review by estesp — APPROVED** [review]

**Pull request merged**